.script/SolutionValidations/validSupportObject.ts

```diff
@@ -0,0 +1,79 @@
+        const filteredResource = resources.filter(function (resource: { type: string; }) {
+            return resource.type === "Microsoft.OperationalInsights/workspaces/providers/metadata";
+        });
+
+        if (filteredResource.length > 0) {
+            filteredResource.forEach((element: { hasOwnProperty: (arg0: string) => boolean; properties: { hasOwnProperty: (arg0: string) => boolean; support: { hasOwnProperty: (arg0: string) => boolean; name: any; email: any; link: any; }; }; }) => {
+                // check if the resource has a "properties" field
+                if (element.hasOwnProperty("properties") === true) {
+                    // check if the "properties" field has a "support" field
+                    if (element.properties.hasOwnProperty("support") === true) {
+                        const support = element.properties.support;
+
+                        if (!support.hasOwnProperty("name")) {
+                            throw new MainTemplateSupportObjectValidationError(`The support object must have a "name" field.`);
+                        } else if (support.name.trim() === "") {
+                            throw new MainTemplateSupportObjectValidationError(`The support object "name" field value cannot be empty.`);
+                        } else if ((!support.hasOwnProperty("email") || support.email.trim() === "") && (!support.hasOwnProperty("link") || support.link.trim() === "")) {
+                            throw new MainTemplateSupportObjectValidationError(`The support object must have either "email" or "link" field and the value should not be empty.`);
+                        } else {
+                            // check if the email is valid
+                            if (support.hasOwnProperty("email") && support.email.trim() !== "") {
+                                const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+                                if (!emailRegex.test(support.email)) {
+                                    throw new MainTemplateSupportObjectValidationError(`Invalid email format for support email: ${support.email}`);
+                                }
+                            }
+
+                            // check if the link is a valid url
+                            if (support.hasOwnProperty("link") && support.link.trim() !== "") {
+                                const linkRegex = /^https?:\/\/\S+$/;
+                                if (!linkRegex.test(support.link)) {
+                                    throw new MainTemplateSupportObjectValidationError(`Invalid url format for support link: ${support.link}`);
+                                }
+                            }
+
+                            // check if either the email or the link field is present and not empty
+                            if ((support.hasOwnProperty("email") && support.email.trim() !== "") || (support.hasOwnProperty("link") && support.link.trim() !== "")) {
```

Comment:
Do we need this check as we already have this in second else if condition?

Reply:
Removed redundant logic.